src/persist-client/src/internal/trace.rs

```diff
@@ -786,14 +786,14 @@ impl<T: Timestamp + Lattice> Spine<T> {
         self.upper.clone_from(batch.upper());
 
         // If `batch` and the most recently inserted batch are both empty,
-        // we can just fuse them. We can also replace a structurally empty
-        // batch with this empty batch, preserving the apparent record count
-        // but now with non-trivial lower and upper bounds.
+        // we can just fuse them.
         if batch.is_empty() {
             if let Some(position) = self.merging.iter().position(|m| !m.is_vacant()) {
                 if self.merging[position].is_single() && self.merging[position].is_empty() {
                     self.insert_at(Some(batch), position);
-                    let merged = self.complete_at(position, log);
+                    let merged = self
+                        .complete_at(position, log)
+                        .expect("just-inserted batch");
```

Comment:
my attempt at this has it mapping `None` to `Vacant` which I slightly prefer to the expect, but this is also fine if you feel strongly

```
                    if let Some(merged) = merged {
                        self.merging[position] = MergeState::Single(merged);
                    } else {
                        self.merging[position] = MergeState::Vacant;
                    }
```

Reply:
Don't feel strongly! (I think we could _probably_ eliminate this case entirely with a larger refactoring, but I was making some effort to avoid changing the API too drastically...)